### DISCUSSION TITLE
[Component: Dialog] How to attach more props to DialogContent components?

### DISCUSSION THREAD

**Discussion by vanirvan** [shadcn-ui/ui#4431] [General]
I don't know how to "attach more props" to these react ref, yes i don't understand ref, so please help me.

so, the original DialogContent component is looks like this:
```tsx
const DialogContent = React.forwardRef<
  React.ElementRef<typeof DialogPrimitive.Content>,
  React.ComponentPropsWithoutRef<typeof DialogPrimitive.Content>
>(({ className, children, ...props }, ref) => (
  <DialogPortal>
    <DialogOverlay />
    <DialogPrimitive.Content
      ref={ref}
      className={cn(
        "fixed left-[50%] top-[50%] z-50 grid w-full max-w-lg translate-x-[-50%] translate-y-[-50%] gap-4 border border-neutral-200 bg-white p-6 shadow-lg duration-200 data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[state=closed]:slide-out-to-left-1/2 data-[state=closed]:slide-out-to-top-[48%] data-[state=open]:slide-in-from-left-1/2 data-[state=open]:slide-in-from-top-[48%] dark:border-neutral-800 dark:bg-neutral-950 sm:rounded-lg",
        className,
      )}
      {...props}
    >
      {children}
      <DialogPrimitive.Close className="absolute right-4 top-4 rounded-sm opacity-70 ring-offset-white transition-opacity hover:opacity-100 focus:outline-none focus:ring-2 focus:ring-neutral-950 focus:ring-offset-2 disabled:pointer-events-none data-[state=open]:bg-neutral-100 data-[state=open]:text-neutral-500 dark:ring-offset-neutral-950 dark:focus:ring-neutral-300 dark:data-[state=open]:bg-neutral-800 dark:data-[state=open]:text-neutral-400">
        <X className="h-4 w-4" />
        <span className="sr-only">Close</span>
      </DialogPrimitive.Close>
    </DialogPrimitive.Content>
  </DialogPortal>
));
DialogContent.displayName = DialogPrimitive.Content.displayName;
/
```

The scenario is, i want to display a dialog that can't be closed except by certain action, mainly for blocking the real content behind the page.

In Radix's Dialog docs, i found `onCloseFocus`, `onEscapeKeyDown`, `onPointerDownOutside`, and `onInteractOutside` options that i can pass on `DialogContent` and it works well, but Shadcn-ui's decide to add `DialogClose` inside `DialogContent` by default.

My current workaround is to duplicate the entire dialog, and the comment `<DialogClose />` inside `DialogContent` on the duplicated components.

**Comment by elistratoov** [comment]
@vanirvan
Hi, i think there is what you need, you can create another custom dialog close

> @yogiyendri Hi! You can separately declare this button and call it whenever you need ))
> 
> ### 1. Declare like this
> <img alt="изображение" width="612" src="https://private-user-images.githubusercontent.com/42961493/281825886-589e9bb3-c821-45aa-bd85-b4296dff9a87.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIyMzMyMTMsIm5iZiI6MTcyMjIzMjkxMywicGF0aCI6Ii80Mjk2MTQ5My8yODE4MjU4ODYtNTg5ZTliYjMtYzgyMS00NWFhLWJkODUtYjQyOTZkZmY5YTg3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI5VDA2MDE1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYxM2IzMjBhM2ExYjQ5ZDY1OTM4NGZlYmM3ZDY3M2I4MzljOTI1ZmE5MDY1MDY2MmFlNjlmNDFhMDhmMjNkMWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.JgQhppyiriJFCkxVL8k9oF5GDT_IWOgnDHe0kws8B7w">
> ### 2. Используйте, когда хотите
> <img alt="изображение" width="528" src="https://private-user-images.githubusercontent.com/42961493/281826257-d1790bf8-5e3a-465a-a27f-0de57920bf75.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIyMzMyMTMsIm5iZiI6MTcyMjIzMjkxMywicGF0aCI6Ii80Mjk2MTQ5My8yODE4MjYyNTctZDE3OTBiZjgtNWUzYS00NjVhLWEyN2YtMGRlNTc5MjBiZjc1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI5VDA2MDE1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNhOGY0YjBjMWYwOGI3ODk3OTkxZTRiNTRkMGFiYzk3MWUwNDFmZWQ1OTNhZDhkN2NiNGUzZTc3ODA4MzlhNmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Wv8UTXGj401g1aa7dJS_FFZS-iaDWWyayLwcqFUFKzs">

#
